### ISSUE TITLE
Diff with an ambiguous branch name fails

### ISSUE THREAD

**Issue by GDYendell** [jesseduffield/lazygit#2764]
**Describe the bug**
Using the diff interface on branch with the same name as a file or directory in the root of the repo gives a git error. The diff command could have ` --` appended to the end to avoid this.

**To Reproduce**

1. Make a new repo
2. Make a file `test` with some content and commit
3. Make a branch test and check it out
4. Make and commit some change to `test`
5. `git diff test main` or equivalent in lazygit gives the error:

```
❯ git diff test main
fatal: ambiguous argument 'test': both revision and filename
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

**Expected behavior**

```
❯ git diff test main --
diff --git a/test b/test
index 317e967..ce01362 100644
--- a/test
+++ b/test
@@ -1,2 +1 @@
 hello
-hello
```

**Version info:**
```
❯ lazygit --version
commit=v0.38.2, build date=2023-05-03T15:21:01Z, build source=binaryRelease, version=0.38.2, os=linux, arch=amd64, git version=2.41.0
❯ git --version
git version 2.41.0
```

**Additional context**

**Note:** please try updating to the latest version or [manually building](https://github.com/jesseduffield/lazygit/#manual) the latest `master` to see if the issue still occurs.

**Comment by GDYendell** [comment]
Actually it looks like this is fixed on master!

**Comment by GDYendell** [comment]
Fixed in 0.40.2 (or earlier).
